### PR TITLE
fix(repo): ensure index generates in the right order

### LINKED ISSUE #1411: Helm generates index in the wrong order
Checking the latest stable index `http://storage.googleapis.com/kubernetes-charts/index.yaml`, the canary build is ordering the versions incorrectly.

### PR DESCRIPTION
**Pull request by prydonius**
fixes #1411

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1412)

### CODE REVIEW

File: pkg/repo/index.go [hunk 1]
```diff
@@ -222,6 +222,7 @@ func IndexDirectory(dir, baseURL string) (*IndexFile, error) {
 		}
 		index.Add(c.Metadata, fname, baseURL, hash)
 	}
+	index.SortEntries()
```
Comment (prydonius) [inline pkg/repo/index.go]: not sure if `IndexDirectory` should be responsible for this?

**Comment by prydonius** [comment]
@technosophos strangely the ordering was correct before #1402 got merged, but I can't see what could have caused that in that PR.

**Comment by technosophos** [comment]
@prydonius I know what happened there. I'm doing a quick look at the codebase to see if it makes more sense to sort here, or to sort in `cmd/helm/repo_index.go`

**Comment by technosophos** [comment]
Okay, I think we should do this as implemented here in the change. Originally, I viewed sort as an extra step, but since we have made a strong comittment regarding ordering, it makes sense to have that commitment consistently enforced.

**Comment by prydonius** [comment]
After discussing over a call, @technosophos decided it would make more sense to explicitly call sort in `cmd/helm/repo_index.go`. It will be more of a pain to consistently enforce an in-memory sorted index, since each method that modified the index would need to call sort.

PTAL

**Comment by technosophos** [comment]
Yeah, this feels like the best direction for now. Thanks for updating all those tests so I don't accidentally do this again.

**Pull request merged**